Sources/Core/StaticString.swift

```diff
@@ -0,0 +1,74 @@
+// Copyright © 2022 Max Desiatov <max@desiatov.com>.
+// All Rights Reserved.
+// SPDX-License-Identifier: BSD-3
+
+@frozen
+public struct StaticString {
+  @usableFromInline
+  internal var _startPtrOrData: Builtin.Word
+
+  @usableFromInline
+  internal var _utf8CodeUnitCount: Builtin.Word
+
+  @usableFromInline
+  internal var _flags: Builtin.Int8
+
+  @_transparent
+  public init() {
+    self = ""
+  }
+
+  @usableFromInline
+  @_transparent
+  internal init(
+    _start: Builtin.RawPointer,
+    utf8CodeUnitCount: Builtin.Word,
+    isASCII: Builtin.Int1
+  ) {
+    _startPtrOrData = Builtin.ptrtoint_Word(_start)
+    _utf8CodeUnitCount = utf8CodeUnitCount
+    _flags = Bool(isASCII)
```

Comment:
This took me a second to understand.  Can you add a comment to the `_flags` ivar to indicate what the bit flags are and what is reserved?